HealthcareSystem/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -54,6 +59,124 @@ private void LoadMapOnCanvas()
 
         private void saveMap()
             => repository.SaveMap(AllMapObjects);
+
+
+
+        private void Canvas_MouseMove(object sender, MouseEventArgs e)
+        {
+            _mouseCurrentPosition = e.GetPosition(this.Canvas);
+           
+            findSelectedMapObject();
+        }
+        
+        private MapObject findSelectedMapObject()
+        {
+            List<MapObject> MapObjectsThatContainMouseCursor = findAllMapObjectsThatContainMouseCursor(AllMapObjects);
+
+            MapObject selectedMapObject = findMapObjectWithMinimumArea(MapObjectsThatContainMouseCursor);
+
+            if (selectedMapObject != null)
+            {
+                selectedMapObject.Rectangle.Effect = new DropShadowEffect { Direction = 0, ShadowDepth = 0, BlurRadius = 14, Opacity = 1, Color = Colors.MediumPurple };
+            }
+
+            return selectedMapObject;
+        }
+
+        private List<MapObject> findAllMapObjectsThatContainMouseCursor(List<MapObject> AllMapObjects)
+        {
+            List<MapObject> MapObjectsThatContainMouseCursor = new List<MapObject>();
+
+            foreach (MapObject mapObject in AllMapObjects)
+            {
+                if (isMouseXCoordinateInsideRectangle(_mouseCurrentPosition.X, mapObject.Rectangle) && isMouseYCoordinateInsideRectangle(_mouseCurrentPosition.Y, mapObject.Rectangle))
+                {
+                    MapObjectsThatContainMouseCursor.Add(mapObject);
+                }
+                else
+                {
+                    mapObject.Rectangle.Effect = null;
+                }
+            }
+
+            return MapObjectsThatContainMouseCursor;
+        }
+
+        private MapObject findMapObjectWithMinimumArea(List<MapObject> MapObjects)
+        {
+            if (MapObjects.Count != 0)
+            {
+                MapObject mapObjectWithMinimumArea = MapObjects[0];
+
+                foreach (MapObject mapObject in MapObjects)
+                {
+                    if ((mapObject.MapObjectArea) < (mapObjectWithMinimumArea.MapObjectArea))
+                    {
+                        mapObjectWithMinimumArea = mapObject;
+                        mapObjectWithMinimumArea.Rectangle.Effect = new DropShadowEffect { Direction = 0, ShadowDepth = 0, BlurRadius = 14, Opacity = 1, Color = Colors.MediumPurple };
+                    }
+                    else
+                    {
+                        mapObject.Rectangle.Effect = null;
+                    }
+                }
+
+                return mapObjectWithMinimumArea;
+            }
+            else
+            {
+                return null;
+            }
+            
+        }
+
+
+        private void Canvas_MouseDown(object sender, MouseButtonEventArgs e)
+        {
+            _mouseCurrentPosition = e.GetPosition(this.Canvas);
+
+            MapObject selectedMapObject = findSelectedMapObject();
```

Comment:
This really points out why I have issues with where the logic for applying shadows on hover is placed - for example, the shadow side effects are happening here as well, even though there's no need.